### PR TITLE
docs: #1362 add incompatible changes to upgrade page

### LINKED ISSUE #1362: add incompatible changes to upgrade page
### Describe the Issue

Add a section to describe incompatible changes with new versions. Right now this is only covered in release notes. Add a section to the Upgrades page in case users are depending on the old behavior.

### Updates to existing documentation

https://docs.redpanda.com/docs/manage/cluster-maintenance/rolling-upgrade/

### Additional notes

Current item:
Starting in version 23.1, if remote read is disabled, and if the requested term falls below the local log's beginning, then Redpanda updates the Kafka handler to return the start offset of Redpanda's local log. In other words, if you request a value older than the lowest offset, Redpanda returns the lowest offset, not -1.

sme = Andrew Wong, Matt

### PR DESCRIPTION
**Pull request by micheleRP**
resolves #1362 

This adds a small section at the end of the Upgrade Prereqs section about incompatible changes. It also moves Upgrade to a top level in the nav tree (no moving files, so no redirects), and updates the index files. 

preview: 
- https://deploy-preview-1800--redpanda-documentation.netlify.app/docs/manage/cluster-maintenance/rolling-upgrade/#prerequisites
- https://deploy-preview-1800--redpanda-documentation.netlify.app/docs/upgrade/migrate/

### CODE REVIEW

**Review by JakeSCahill — APPROVED** [review]

File: docs/manage/cluster-maintenance/rolling-upgrade.mdx [hunk 1]
```diff
@@ -35,6 +35,12 @@ Redpanda platform version numbers follow the convention AB.C.D, where AB is the
 - A running Redpanda cluster.
 - [jq](https://stedolan.github.io/jq/download/) for listing available versions.
 - An understanding of the [impact of broker restarts](#impact-of-broker-restarts) on clients, node CPU, and any alerting systems you use.
+- Review incompatible changes in new versions:
+
+  | Version | Change | 
+  | ---     | ---    |
+  |23.1     | If remote read is disabled, and if the requested term falls below the local log's beginning, then Redpanda updates the Kafka handler to return the start offset of Redpanda's local log. In other words, if you request a value older than the lowest offset, Redpanda returns the lowest offset, not -1. |
```
Comment (mattschumpert) [inline docs/manage/cluster-maintenance/rolling-upgrade.mdx:42]: I suggest starting with just the 'in other words' part. This is actually what the customer deals with.

@andrwng what do you think? Cant we just skip the whole first sentence (from a user perspective)?

File: docs/manage/kubernetes/rolling-upgrade.mdx [hunk 2]
```diff
@@ -35,6 +35,11 @@ Redpanda platform version numbers follow the convention AB.C.D, where AB is the
 - A running Redpanda cluster.
 - [jq](https://stedolan.github.io/jq/download/) for listing available versions.
 - An understanding of the [impact of broker restarts](#impact-of-broker-restarts) on clients, node CPU, and any alerting systems you use.
+- Review incompatible changes in new versions:
+
+  | Version | Change | 
+  | ---     | ---    |
+  |23.1     | If remote read is disabled, and if the requested term falls below the local log's beginning, then Redpanda updates the Kafka handler to return the start offset of Redpanda's local log. In other words, if you request a value older than the lowest offset, Redpanda returns the lowest offset, not -1. |
```
Comment (mattschumpert) [inline docs/manage/kubernetes/rolling-upgrade.mdx:42]: Same comment as above

**Review by mattschumpert — APPROVED** [review]

File: docs/manage/kubernetes/rolling-upgrade.mdx (hunk 2 above)
Comment (micheleRP) [reply docs/manage/kubernetes/rolling-upgrade.mdx:42]: @mattschumpert please confirm which version we should go with: 

Patches in [22.3.14](https://github.com/redpanda-data/redpanda/discussions/9522) and [23.1.2](https://github.com/redpanda-data/redpanda/discussions/9523) changed the behavior when remote read is disabled and the requested term falls below the local log's beginning. In earlier versions, Redpanda returned an offset -1. With the patch, when you request a value older than the lowest offset, Redpanda returns the lowest offset, not -1.

OR

Patches in [22.3.14](https://github.com/redpanda-data/redpanda/discussions/9522) and [23.1.2](https://github.com/redpanda-data/redpanda/discussions/9523) changed the behavior when you request a value older than the lowest offset. In earlier versions, Redpanda returned an offset -1. With the patch, Redpanda returns the lowest offset, not -1.

Comment (andrwng) [reply docs/manage/kubernetes/rolling-upgrade.mdx:42]: I'd pick the first one; the distinction with local data and the point about remote reads are important.

File: docs/manage/cluster-maintenance/rolling-upgrade.mdx (hunk 1 above)
Comment (andrwng) [reply docs/manage/cluster-maintenance/rolling-upgrade.mdx]: I'm not sure how much detail you want in these notes, but IMO it's important to characterize the situations the bug happens so readers know whether they should care. The first sentence establishes that.

**Review by andrwng — APPROVED** [review]

**Pull request merged**